### PR TITLE
54 - introduced react to confidence slider

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt-contrib-confidenceSlider/issues/54

### CODE REVIEW

**Review by ethan-lp — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by lc-alexanderbenesch — APPROVED** [review]
👍

**Review by joe-allen-89 — APPROVED** [review]
👍

File: templates/confidenceSlider.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    _isCorrect,
+    labelStart,
+    labelEnd,
+    _userAnswer,
+    _scaleStart,
+    _scaleEnd,
+    _marginDir
+  } = props;
+
+  const shouldShowMarking = isInteractive() && _canShowMarking;
```
Comment (oliverfoster) [inline templates/confidenceSlider.jsx:20]: https://github.com/adaptlearning/adapt-contrib-core/blob/759b768057ef3d88d83328121fdec25f1a96bb93/js/models/questionModel.js#L368-L374

You should be able to use `_shouldShowMarking` directly from the model now.

**Review by oliverfoster — CHANGES_REQUESTED** [review]

**Comment by oliverfoster** [comment]
@eleanor-heath Could you see if this has an easy fix whilst you're here? https://github.com/adaptlearning/adapt_framework/issues/3238

**Comment by eleanor-heath** [comment]
> @eleanor-heath Could you see if this has an easy fix whilst you're here? [adaptlearning/adapt_framework#3238](https://github.com/adaptlearning/adapt_framework/issues/3238)
will do

File: js/confidenceSliderView.js [hunk 2]
```diff
@@ -94,7 +94,12 @@ export default class ConfidenceSliderView extends SliderView {
   }
 
   onLinkedSubmittedChanged(linkedModel) {
+    this.$('.component__body-inner').html(linkedModel.get('body'));
```
Comment (oliverfoster) [inline js/confidenceSliderView.js:97]:
```suggestion
    this.$('.component__body-inner').html(this.model.get('body'));
```
This should probably be the body from the model for this instance and not the linked instance.

**Pull request merged**